EducationalAssignments/ImmutableVersionOne.md

```diff
@@ -0,0 +1,262 @@
+
+```bash
+python2 repy.py restrictions.default encasementlib.r2py [security_layer].r2py [application].r2py
+```
+
+(Replace `[security_layer].r2py` and `[application].r2py` by the names of the security layers and application that you want to run.)
+
+In order to test whether or not these steps worked, please copy and paste the code found below for the sample security layer and sample attack.  
+
+You should not get any errors (or outputs) when running the test. If not, please go through the troubleshooting section found below.
+
+### Troubleshooting Repy code
+---
+
+If you can't get Repy files to run, some of the following common errors may have occurred:
+
+- **using `print` instead of `log`:**  
+  Repy is a subset of Python, but its syntax is slightly different. For example, Python's `print` statement cannot be used; Repy has `log` for that. For a full list of acceptable syntax please see  
+  <https://github.com/SeattleTestbed/docs/blob/master/Programming/RepyV2API.md>
+
+- **command line errors:**
+
+  **files are missing:** In the above command line call, you must have `repy.py`, `restrictions.default`, `encasementlib.r2py`, the security layer and the program you want to run in the current working directory. If any or all of the above files are not in that directory then you will not be able to run repy files.
+
+### Tutorials for Repy and Python
+----
+
+Now that you have Repy and Python, you may need a refresher on how to use them.
+The following tutorials provide this information.
+
+ * Official [Python tutorial](http://docs.python.org/tutorial/)
+ * [Differences between RepyV2 and Python](../Programming/PythonVsRepyV2.md)
+ * List of [RepyV2 API calls](../Programming/RepyV2API.md)
+
+## Building the security layer   
+------
+
+The following program is a sample security layer, it is not complete and does not handle all cases required by the API. Remember, you have no idea how the attacker will try to penetrate your security layer, so it is important that you leave nothing to chance!
+
+#### A basic (and inadequate) defense
+
+Time to start coding! Let's inspect a basic security layer. You can use the code given below as a template for your reference monitor.  
+You may save the file as `reference_monitor_[netid].r2py`.
+
+```py
+"""
+This security layer inadequately handles the Versioned and Immutable functionality
+
+Note:
+    This security layer uses encasementlib.r2py, restrictions.default, repy.py and Python
+    Also you need to give it an application to run.
+    python repy.py restrictions.default encasementlib.r2py [security_layer].r2py [attack_program].r2py 
+    
+"""
+TYPE = "type"
+ARGS = "args"
+RETURN = "return"
+EXCP = "exceptions"
+TARGET = "target"
+FUNC = "func"  
+OBJC = "objc"
+
+
+class VMFile():
+    def __init__(self, filename, create):
+        if create:
+            self.VMfile = openfile(filename, True)
```

Comment:
at least see if the file exists with listfiles and then create a version, etc.  A little more functionality here would be useful.